src/libraries/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/PkcsPalWindows.Encrypt.cs

```diff
@@ -153,7 +153,7 @@ public static SafeCryptMsgHandle CreateCryptMsgHandleToEncode(CmsRecipientCollec
                     DATA_BLOB* pCertEncoded = (DATA_BLOB*)(hb.Alloc(numCertificates, sizeof(DATA_BLOB)));
                     for (int i = 0; i < numCertificates; i++)
                     {
-                        byte[] certEncoded = originatorCerts[i].Export(X509ContentType.Cert);
+                        byte[] certEncoded = originatorCerts[i].Export(X509ContentType.Cert)!;
```

Comment:
In what situation will Export return null, and what guarantees that won't happen here?

Reply:
Actually, I think we messed up X509Certificate.Export.  It "looks" nullable from the code, because it ends up in a common path with collection export.  `null` only happens for exporting an empty collection as Cert or SerializedCert.  (Why? Because that's what someone decided in net20, before my time at the helm :smile:).

Reply:
I don't know if it makes sense to fix the X509Certificate.Export overloads as non-nullable in this PR, or a different one.

* X509Certificate.Export overloads
* ICertificatePalCore.Export
  * When the Windows one unifies with the IStorePal.Export it should just assert the return isn't null.

Reply:
I think a separate one, and this can rebase on that.  Mind submitting one? :)

Reply:
Sure... working on it now.

Reply:
I can update that with this PR, its returning null when all certs read and end of the store has been reached, that means exporting within this loop is null safe, but not sure if it always be guaranteed that any call will be made within certificate count, if that is the case what we want to do with the below implementation of Export on explicit null return? Bang?
```csharp
public byte[]? Export(X509ContentType contentType, SafePasswordHandle password)
{
    Debug.Assert(password != null);
    switch (contentType)
    {
        case X509ContentType.Cert:
        {
            SafeCertContextHandle? pCertContext = null;
            if (!Interop.crypt32.CertEnumCertificatesInStore(_certStore, ref pCertContext))
                return null;
            try
                {
                    unsafe
                    {
                        byte[] rawData = new byte[pCertContext.CertContext->cbCertEncoded];
                        Marshal.Copy((IntPtr)(pCertContext.CertContext->pbCertEncoded), rawData, 0, rawData.Length);
                        GC.KeepAlive(pCertContext);
                        return rawData;
                    }
                }
            finally
            {
                pCertContext.Dispose();
            }
        }
        case X509ContentType.SerializedCert:
        {
            SafeCertContextHandle? pCertContext = null;
            if (!Interop.crypt32.CertEnumCertificatesInStore(_certStore, ref pCertContext))
                return null;

            try
            {
                 ...
            }
     ...
```

Reply:
#32625

Reply:
Thank you @bartonjs

Reply:
My update to X509Certificates is now merged; so this bang can (has to?) go away now after you pull master in again.

Reply:
>  so this bang can (has to?)

Can, but not has to.  You can have !s in places they're not needed and the compiler won't warn.  I'm sure we already have some, as the compiler has improved its analysis and eliminated warnings in places we were previously using !s to suppress them.  There's a desire to have an analyzer with auto-fix support to remove unneeded !s... hopefully that'll emerge at some point :)

Reply:
> My update to X509Certificates is now merged; so this bang can (has to?) go away now after you pull master in again.

Pulled master and removed the bang